AX25_print()/NETROM_print() overflow buff[8] on callsigns with a two-digit SSID
https://github.com/ecki/net-tools/commit/c084d1fea5de0f6dcaed4a59b38a4140bd2e9f13
https://github.com/ecki/net-tools/commit/139f5d85e4e93bd75bc1072349bce19bf56c058a

diff --git a/lib/ax25.c b/lib/ax25.c
index 80a82c4..aee4214 100644
--- a/lib/ax25.c
+++ b/lib/ax25.c
@@ -47,9 +47,10 @@ static char AX25_errmsg[128];
 
 extern struct aftype ax25_aftype;
 
+// align with NETROM_orint
 static const char *AX25_print(const char *ptr)
 {
-    static char buff[8];
+    static char buff[10]; // N0CALL-15
     int i;
 
     for (i = 0; i < 6; i++) {
@@ -58,9 +59,14 @@ static const char *AX25_print(const char *ptr)
 	    buff[i] = '\0';
     }
     buff[6] = '\0';
+
+    // add SSID
     i = ((ptr[6] & 0x1E) >> 1);
-    if (i != 0)
-	sprintf(&buff[strlen(buff)], "-%d", i);
+    if (i != 0) {
+        int l = strlen(buff);
+        snprintf(&buff[l], sizeof(buff)-l, "-%d", i);
+    }
+
     return (buff);
 }
 
diff --git a/lib/netrom.c b/lib/netrom.c
index 6bcde2d..309e7cb 100644
--- a/lib/netrom.c
+++ b/lib/netrom.c
@@ -54,7 +54,7 @@ extern struct aftype netrom_aftype;
 
 static const char *NETROM_print(const char *ptr)
 {
-    static char buff[8];
+    static char buff[10]; // N0CALL-15\0
     int i;
 
     for (i = 0; i < 6; i++) {
@@ -63,9 +63,15 @@ static const char *NETROM_print(const char *ptr)
 	    buff[i] = '\0';
     }
     buff[6] = '\0';
+
+    // add SSID
     i = ((ptr[6] & 0x1E) >> 1);
     if (i != 0)
-	sprintf(&buff[strlen(buff)], "-%d", i);
+    {
+        int l = strlen(buff); // 0-6
+        snprintf(&buff[l],sizeof(buff)-l, "-%d", i);
+    }
+
     return (buff);
 }
 
